src/main/java/org/jmxtrans/agent/RollingFileOutputWriter.java

```diff
@@ -94,7 +100,11 @@ public void writeInvocationResult(String invocationName, Object value) throws IO
 
     public synchronized void writeQueryResult(@Nonnull String name, @Nullable String type, @Nullable Object value) throws IOException {
         try {
-            getTemporaryFileWriter().write("["+dfISO8601.format(Calendar.getInstance().getTime()) +"] "+name + " " + value + "\n");
+            if (singleLine) {
+                getTemporaryFileWriter().write(name + "=" + value + ", ");
```

Comment:
Won't this cause a trailing comma on each line? To avoid that we could save all metrics in `writeQueryResult` and do the actual write in `postCollect`